### PR TITLE
Documentation for the visualizations

### PR DESCRIPTION
**Pull request by bouthilx**
[Solve #527 #495]

The documentation for this branch is automatically built and available at https://orion.readthedocs.io/en/doc-viz/

See visualization examples at https://orion.readthedocs.io/en/doc-viz/auto_examples/index.html
And tutorials at https://orion.readthedocs.io/en/doc-viz/auto_tutorials/index.html

## Work in progress
(checked when done and ready for review)

Viz Examples
- [x] Regret curves
- [x] Parallel Coordinates
- [x] Local Parameter Importance
- [x] Partial Dependencies

How-tos
- [x] How to plot
- [x] How to save
- [x] How to share on chart studio

Tutorials
- [x] Python API basics
- [x] Checkpointing trials
- [ ] Benchmarking

## Reviewed and approved 

Viz Examples
- [ ] Regret curves
- [ ] Parallel Coordinates
- [ ] Local Parameter Importance
- [ ] Partial Dependencies

How-tos
- [ ] How to plot
- [ ] How to save
- [ ] How to share on chart studio

Tutorials
- [ ] Python API basics
- [ ] Checkpointing trials
- [ ] Benchmarking

### CODE REVIEW

File: examples/tutorials/code_2_hyperband_checkpoint.py [hunk 1]
```diff
@@ -0,0 +1,392 @@
...
+using the command line API. To this end, your script should accept an argument for the checkpoint
+file path. Suppose this argument is ``--checkpoint``, you should call your script with the
+following template.
+
+::
+
+    orion hunt -n <exp name>
+        ./your_script.sh --checkpoint '{experiment.working_dir}/{trial.hash_params}'
+
+Your script it reponsible to take this checkpoint path, resume from checkpoints or same
```
Comment (wagnersj) [inline examples/tutorials/code_2_hyperband_checkpoint.py:34]: Grammar/Spelling: "Your script it responsible..." -> "Your script is responsible..."

File: examples/tutorials/code_2_hyperband_checkpoint.py [hunk 2]
```diff
@@ -0,0 +1,392 @@
...
+#     :file: ../_static/hyperband-cifar10_partial_dependencies_params.html
+#
+# The main culprit for the high validation error rate seems to be the wide prior for ``gamma``.
+# Because of this Hyperband spent most of the computation time on bad ``gamma``s. This prior
+# should be narrowed to ``uniform(0.995, 1)``.
+# The prior for the learning rate could also be narrowed to ``loguniform(0.001, 0.1)`` to
+# help the optimization.
+#
+# Note that Hyperband could also find better results without adjusting the search space, but
+# it would required significantly more repetitions.
```
Comment (wagnersj) [inline examples/tutorials/code_2_hyperband_checkpoint.py:392]: At end of rendered HTML file, the text "Total running time of the script: ( 0 minutes 0.000 seconds)" appears, but not sure what this is all about. It also appears in all of the visualization doc pages.

File: examples/plotting/plot_1_regret.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+=============
+Regret curves
+=============
+
+.. hint:: 
+
+   Conveys how quickly the algorithm found the best hyperparameters.
+    
+The regret is the difference between the achieved objective and the optimal objective.
+Plotted as a time serie, it shows how fast an optimization algorithm approaches the best
```
Comment (wagnersj) [inline examples/plotting/plot_1_regret.py:11]: Spelling: "time serie" -> "time series"

File: examples/tutorials/code_2_hyperband_checkpoint.py (hunk 2 above)
Comment (bouthilx) [reply examples/tutorials/code_2_hyperband_checkpoint.py:392]: I'll see if there is an option to remove this. It's part of the sphinx-gallery, when it executes the scripts.

File: examples/plotting/plot_3_lpi.py [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+
+# Load the data for the specified experiment
+experiment = get_experiment("2-dim-exp", storage=storage)
+fig = experiment.plot.lpi()
+fig
+
+#%%
+# On this plot the x-axis shows the different hyperparameters while the y-axis gives the
+# local parameter importance.
+# The error bars represents the standard deviation of the LPI based on 10 runs. Remember
```
Comment (wagnersj) [inline examples/plotting/plot_3_lpi.py:42]: Grammer: "The error bars represents..." -> "The error bars represent..."

File: examples/plotting/plot_3_lpi.py [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+# (10 times by default) using different random seeds. Here is an example
+# setting the number of points for the grids, the number of runs and the initial random seed
+# for the regression model.
+
+experiment.plot.lpi(n_points=10, n_runs=5, model_kwargs=dict(random_state=1))
+
+#%%
+# We can see that the learning rate had a larger impact than the dropout in achieving the best
+# objective.
+# A search space of only 2 dimensions is easy to analyse visualy however so the LPI
```
Comment (wagnersj) [inline examples/plotting/plot_3_lpi.py:54]: Spelling: "visualy" -> "visually"

File: examples/plotting/plot_3_lpi.py [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+# There is a large difference here between the most important hyperparameter (learning rate) and the
+# least important one (gamma).
+#
+# One caveat of LPI is that the variance for each hyperparameters depends on the search space.
+# If the prior for one hyperparameter is narrow and fits the region of best values for this
+# hyperparameter, then the variance will be low and this hyperparameter will be considered
+# non-important. It may be important, but it is not important to optimize it within this narrow
+# search space. Another related issue, is that if one hyperparameter have a dramatic effect, it will
+# lead to a variance so large that the other hyperparameters will seem unrelevant in comparison.
+# This is what we observe here with the learning rate. If we branche from the experiment and define
```
Comment (wagnersj) [inline examples/plotting/plot_3_lpi.py:73]: Spelling: "branche" -> "branch"

File: examples/plotting/plot_3_lpi.py [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+        "epochs": "fidelity(1, 120, base=4)",
+        "learning_rate": "loguniform(1e-3, 0.1)",
+        "momentum": "uniform(0, 0.9)",
+        "weight_decay": "loguniform(1e-10, 1e-2)",
+        "gamma": "loguniform(0.97, 1)",
+    },
+    storage=storage,
+)
+
+experiment.plot.lpi()
```
Comment (wagnersj) [inline examples/plotting/plot_3_lpi.py:97]: No data appears to be displayed in the plot.

File: examples/plotting/plot_4_partial_dependencies.py [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+# ~~~~~~
+#
+# The simple example involved only 2 hyperparameters, but typical search spaces can be much larger.
+# The partial dependency plot becomes hard to read with more than 3-5 hyperparameters dependency on
+# the size of your screen. With a fix width like in this documentation, 5 hyperparameters are
+# impossible to read as you can see below.
+# (Data coming from tutorial :ref:`sphx_glr_auto_tutorials_code_2_hyperband_checkpoint.py`)
+
+experiment = get_experiment("hyperband-cifar10", storage=storage)
+experiment.plot.partial_dependencies()
```
Comment (wagnersj) [inline examples/plotting/plot_4_partial_dependencies.py:87]: The plot generated here has several data points represented by dots, which makes the plots look quite busy and obscures the underlying contours. Is it possible to optionally remove the dots representing the data points?

File: examples/plotting/plot_2_parallel_coordinates.py [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+The parallel coordinates plot decomposes a search space of `n` dimensions into `n`
+axis so that the entire space can be visualized simultaneously. Each dimension
+is represented as a vertical axis and trials are represented as lines crossing each 
+axis at the corresponding value of the hyperparameters. There is no obvious optimal ordering
+for the vertical axis, and you will often find that changing the order helps better understanding
+the data. Additionaly, the lines are plotted with graded colors based on the objective. The
+gradation is shown in a color bar on the right of the plot. Note that the objectives are added
+as the last axis is the plot as well.
+
+.. autofunction:: orion.plotting.base.parallel_coordinates
```
Comment (wagnersj) [inline examples/plotting/plot_2_parallel_coordinates.py:20]: There is a broken link to the plotly API docs in the generated document. Need to fix in source code for referenced function.

File: examples/tutorials/code_2_hyperband_checkpoint.py (hunk 2 above)
Comment (bouthilx) [reply examples/tutorials/code_2_hyperband_checkpoint.py:392]: I could not find an option to remove it.

File: examples/plotting/plot_4_partial_dependencies.py (hunk 8 above)
Comment (bouthilx) [reply examples/plotting/plot_4_partial_dependencies.py:87]: I could change the size of the dots, or make them optional.

File: examples/plotting/plot_3_lpi.py (hunk 7 above)
Comment (bouthilx) [reply examples/plotting/plot_3_lpi.py:97]: Seems ok on my side in current version. I will leave this one as unresolved and double-check if the plot is fine when I push my modifications.

**Pull request merged**